genie-common/src/main/java/com/netflix/genie/common/util/GenieDateFormat.java

```diff
@@ -17,37 +17,52 @@
  */
 package com.netflix.genie.common.util;
 
-import com.fasterxml.jackson.databind.util.ISO8601DateFormat;
-import com.fasterxml.jackson.databind.util.ISO8601Utils;
+import com.fasterxml.jackson.databind.util.StdDateFormat;
 
 import java.text.FieldPosition;
+import java.util.Calendar;
 import java.util.Date;
 import java.util.GregorianCalendar;
 import java.util.TimeZone;
 
 /**
- * An extension of the ISO8601DateFormat to include milliseconds.
+ * The date format used by Genie.
+ * Override the serialization to output a date string containing milliseconds and string representation of the timezone.
  *
  * @author tgianos
  * @since 3.0.0
  */
-public class GenieDateFormat extends ISO8601DateFormat {
+public class GenieDateFormat extends StdDateFormat {
+
+    private static final TimeZone TIMEZONE = TimeZone.getTimeZone("UTC");
 
     /**
      * Constructor.
      */
     public GenieDateFormat() {
         super();
-        this.calendar = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public StringBuffer format(final Date date, final StringBuffer toAppendTo, final FieldPosition fieldPosition) {
-        final String value = ISO8601Utils.format(date, true);
-        toAppendTo.append(value);
-        return toAppendTo;
+        // StdDateFormat and ISO8601Utils do not offer a way to format consistent with the current Genie API format
```

Comment:
I think it would be much simpler to convert the `Date` to an `Instant` and then use [Instant#toString()](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#toString--)

Reply:
Instant format does not include milliseconds :-(

Reply:
Yeah I noticed that. Could still try to take advantage of [DateTimeFormatter](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html) if you want instead of all the Calendar stuff. Up to you.

Reply:
@tgianos tried `DateTimeFormatter` but it's not simpler than the `format` below, comes out similar to this in terms of lines of code, but it's more complex to understand, so I'd leave this as-is. 
Not great that we have to manually provide a format string, but at least it's straighforward what's going on.